rust/settings/BUILD.bazel

```diff
@@ -99,6 +99,19 @@ incompatible_flag(
     issue = "https://github.com/bazelbuild/rules_rust/issues/2324",
 )
 
+# A flag to control whether to link libstd dynamically.
+bool_flag(
+    name = "experimental_use_dylib_linkage",
```

Comment:
I'm not quite sure about the full impact of this change - is it _just_ on the stdlib (in which case I'd expect stdlib to be in the flag name), or is it for all dependencies?

Reply:
I agree. It's only dynamically linking libstd. I'll fix the naming and add tests.

Per https://doc.rust-lang.org/reference/linkage.html, it's worth nothing

> For most situations, having all libraries available as a dylib is recommended if dynamically linking. 

So ideally, I think it makes sense to switch `rust_library` to build `dylib` in this mode. But that's a future :) .